### PR TITLE
254838: Updates to infra helm chart and appConfig

### PR DESCRIPTION
**Pull request by jeevankuduvaravindran**
Updates to infra helm chart and appConfig

[AB#254838](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/254838)

### CODE REVIEW

File: helm/ffc-demo-claim-service-infra/values.yaml [hunk 1]
```diff
@@ -1,23 +1,45 @@
...
+    collation: en_US.utf8
+
+tags:
+  Environment: SND1
+  ServiceCode: ADP
+  ServiceName: ffc-demo-claim-service
+  ServiceType: Shared
+  Purpose: Define purpose
+  ManagedBy: ASO
+  kubernetes_cluster: SNDADPINFAK1401
```
Comment (DaRoszko) [inline helm/ffc-demo-claim-service-infra/values.yaml:38]: should be var?

Comment (DaRoszko) [reply helm/ffc-demo-claim-service-infra/values.yaml:38]: 'snd' values

File: helm/ffc-demo-claim-service-infra/values.yaml [hunk 2]
```diff
@@ -1,23 +1,45 @@
...
+
+#Dummy placeholder values only for linting. NOT TO BE POPULATED
+subscriptionId: "dummySubscriptionId"
```
Comment (DaRoszko) [inline helm/ffc-demo-claim-service-infra/values.yaml:41]: these dummy values can be removed as per @ganeshg2491  coomment

File: helm/ffc-demo-claim-service-infra/values.yaml [hunk 3]
```diff
@@ -1,23 +1,45 @@
...
   - name: calculation
+    roleAssignments:
+      - roleName: 'QueueSender'
 
 namespaceTopics:
   - name: schedule   
+    roleAssignments:
+      - roleName: 'TopicSender'
 
 postgresResourceGroupName: sndadpdbsrg1401
```
Comment (DaRoszko) [inline helm/ffc-demo-claim-service-infra/values.yaml:23]: should be var or abstracted/defaulted?

Comment (jeevankuduvaravindran) [reply helm/ffc-demo-claim-service-infra/values.yaml:23]: These need to be explicitly provided by developer. In the backend on ASO helm library, we will assign proper role assignments based on the role provided here.

File: helm/ffc-demo-claim-service-infra/values.yaml [hunk 4]
```diff
@@ -1,23 +1,45 @@
...
+  ServiceName: ffc-demo-claim-service
+  ServiceType: Shared
+  Purpose: Define purpose
+  ManagedBy: ASO
+  kubernetes_cluster: SNDADPINFAK1401
+
+#Dummy placeholder values only for linting. NOT TO BE POPULATED
+subscriptionId: "dummySubscriptionId"
+userAssignedIdentity: {}
+teamMIPrefix: "dummy-mi-prefix"
```
Comment (ganeshg2491) [inline helm/ffc-demo-claim-service-infra/values.yaml:43]: @jeevankuduvaravindran : Can you remove this dummyvalue variables. It is not required here.

File: helm/ffc-demo-claim-service-infra/values.yaml (hunk 2 above)
Comment (jeevankuduvaravindran) [reply helm/ffc-demo-claim-service-infra/values.yaml:41]: @DaRoszko working on a fix with Ganesh to remove all variables that doesn't concern developer. In progress.

File: helm/ffc-demo-claim-service-infra/values.yaml (hunk 4 above)
Comment (jeevankuduvaravindran) [reply helm/ffc-demo-claim-service-infra/values.yaml:43]: Yeah will remove it as part of the other variables too.

File: helm/ffc-demo-claim-service-infra/values.yaml (hunk 3 above)
Comment (DaRoszko) [reply helm/ffc-demo-claim-service-infra/values.yaml:23]: can you discuss with ganesh how to abstract? they need to be vars too for example as it's env specific

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: helm/ffc-demo-claim-service-infra/values.yaml (hunk 4 above)
Comment (jeevankuduvaravindran) [reply helm/ffc-demo-claim-service-infra/values.yaml]: done

File: helm/ffc-demo-claim-service-infra/values.yaml (hunk 2 above)
Comment (jeevankuduvaravindran) [reply helm/ffc-demo-claim-service-infra/values.yaml]: done

File: helm/ffc-demo-claim-service-infra/values.yaml (hunk 1 above)
Comment (jeevankuduvaravindran) [reply helm/ffc-demo-claim-service-infra/values.yaml]: @DaRoszko removed all variables that doesnt concern developer

File: helm/ffc-demo-claim-service-infra/values.yaml (hunk 3 above)
Comment (jeevankuduvaravindran) [reply helm/ffc-demo-claim-service-infra/values.yaml]: @DaRoszko done

**Review by kbabigumira — DISMISSED** [review]

File: package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ffc-demo-claim-service",
   "description": "Digital service mock to claim public money in the event property subsides into mine shaft.",
-  "version": "5.1.22",
+  "version": "5.1.25",
```
Comment (kbabigumira) [inline package.json:4]: shouldn't this be 5.1.23?

**Review by kbabigumira — DISMISSED** [review]

**Review by DaRoszko — DISMISSED** [review]

**Review by DaRoszko — APPROVED** [review]

File: .azuredevops/build.yaml [hunk 6]
```diff
@@ -53,4 +53,8 @@ extends:
       testFilePath: "./docker-compose.test.yaml"
       snykTestFilePath: "./docker-compose.snyk.yaml"
     appDeployConfig:
-      filepath: "./appConfig"  
\ No newline at end of file
+      filepath: "./appConfig"
+      variableGroups: 
+        - ffc-demo-claim-service-snd1
```
Comment (kbabigumira) [inline .azuredevops/build.yaml:58]: this should not be hardcoded

**Review by ganeshg2491 — APPROVED** [review]

**Review by kbabigumira — APPROVED** [review]

File: appConfig/appConfig.snd1.json [hunk 7]
```diff
@@ -1,109 +1,114 @@
...
+    {
+        "key": "POSTGRES_SCHEMA_NAME",
+        "value": "public",
+        "label": "{{serviceName}}",
+        "content_type": "",
+        "tags": {}
+    },
+    {
+        "key": "APPINSIGHTS_INSTRUMENTATIONKEY",
+        "value": "36932cc4-12b7-4f0f-bbee-36ba20174390",
```
Comment (kbabigumira) [inline appConfig/appConfig.snd1.json:81]: should not be checked in.

File: appConfig/appConfig.snd1.json [hunk 8]
```diff
@@ -1,109 +1,114 @@
...
+    {
+        "key": "POSTGRES_USERNAME",
+        "value": "sndadpinfmid1401-ffc-demo-services",
+        "label": "{{serviceName}}",
+        "content_type": "",
+        "tags": {}
+    },
+    {
+        "key": "NOTIFY_EMAIL_TEMPLATE_KEY",
+        "value": "c048c3e6-d189-45c2-bdf4-ee4a7a732167",
```
Comment (kbabigumira) [inline appConfig/appConfig.snd1.json:95]: Is this sensitive?

File: appConfig/appConfig.snd1.json (hunk 7 above)
Comment (jeevankuduvaravindran) [reply appConfig/appConfig.snd1.json]: @kbabigumira has been removed

**Review by kbabigumira — APPROVED** [review]

**Pull request merged**